File: java/src/main/java/mx/iteso/adapter/googledocs/GoogleDoc.java
package mx.iteso.adapter.googledocs;

import mx.iteso.adapter.BackgroundImage;

import java.awt.Font;
import mx.iteso.adapter.Format;

/**
 * Google Doc Class.
 */
public class GoogleDoc implements IGoogleDoc {
    /**
     * Font.
     */
    private Font font;
    /**
     * Format.
     */
    private Format format;
    /**
     * Bkground image.
     */
    private BackgroundImage backgroundImage;
    /**
     * shared Permission //denied.
     */
    private int sharingPermissions;

    /**
     *
     * @return font.
     */
    public Font getFont() {
        return this.font;
    }

    /**
     *
     * @return Format.
     */
    public Format getStyle() {
        return this.format;
    }

    /**
     *
     * @return Format???????.
     */
    public Format getFormat() {
        return this.format;
    }

    /**
     *
     * @return background img.
     */
    public BackgroundImage getBackground() {
        return this.backgroundImage;
    }

    /**
     *
     * @param incomingSharingPermissions .
     */
    public void setSharingPermissions(final int incomingSharingPermissions) {
        this.sharingPermissions = incomingSharingPermissions;
    }

    /**
     *
     * @return permission denied.
     */
    public int getSharingPermissions() {
        return 0;
    }
}

File: csharp/AdapterPractice/AdapterPractice/Adapter/GoogleDocs/GoogleDoc.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AdapterPractice.Adapter.GoogleDocs
{
    public class GoogleDoc : IGoogleDoc {

    private Font font;

    private Format format;

    private BackgroundImage backgroundImage;

    private int sharingPermissions;


    public Font getFont()
    {
        return this.font;
    }


    public Format getStyle()
    {
        return this.format;
    }

    public Format getFormat()
    {
        return this.format;
    }

    public BackgroundImage getBackground()
    {
        return this.backgroundImage;
    }

    public void setSharingPermissions(int incomingSharingPermissions)
    {
        this.sharingPermissions = incomingSharingPermissions;
    }

    public int getSharingPermissions()
    {
        return 0;
    }
    }

}
